lib/plug/conn.ex

```diff
@@ -460,12 +460,15 @@ defmodule Plug.Conn do
   otherwise `{:error, reason}`.
   """
   @spec chunk(t, body) :: {:ok, t} | {:error, term} | no_return
-  def chunk(%Conn{state: :chunked} = conn, ""), do: {:ok, conn}
   def chunk(%Conn{adapter: {adapter, payload}, state: :chunked} = conn, chunk) do
-    case adapter.chunk(payload, chunk) do
-      :ok                  -> {:ok, conn}
-      {:ok, body, payload} -> {:ok, %{conn | resp_body: body, adapter: {adapter, payload}}}
-      {:error, _} = error  -> error
+    case IO.iodata_length(chunk) do
```

Comment:
@myronmarston `IO.iodata_length/1` is quite fast operation: `876` microseconds to get size of `List.duplicate([List.duplicate([String.duplicate("1", 5), 1, 2], 100), 3, String.duplicate("2", 50)], 1000)` (which is `751_000` bytes).
Not sure about the garbage, but I assume it's handled in C-land. 🤔